Implement update user

oemel09 opened the pull request:
Added put method to update users
CORS changes are also in here...

fabricioflores commented on manage-people/routes/PeopleRoutes.js:
```diff
@@ -50,9 +50,25 @@ const peopleRoutes = (app) => {
+                authentication_identity: req.body.authentication_identity
+            }, function(err) {
+                if(err) {
+                   return res.status(500).send({message: 'Doh...'});
```
you can return a 422 error, and the `err` value that you are getting from the db. 500 is too generic, and also in the frontend you won't see the cause of the error.

fabricioflores commented on manage-people/routes/PeopleRoutes.js:
```diff
@@ -50,9 +50,25 @@ const peopleRoutes = (app) => {
+                if(err) {
+                   return res.status(500).send({message: 'Doh...'});
+                } else {
+                  res.status(200).send();
```
also, maybe you want to return the updated `Person`

fabricioflores commented on app.js:
```diff
@@ -4,10 +4,16 @@ import express from 'express'
+
+app.use(function(req, res, next) {
+    res.header("Access-Control-Allow-Origin", "*");
+    res.header("Access-Control-Allow-Methods", "POST, GET, DELETE,  PUT");
```
I remember I had an issue with `OPTIONS` method. Just if you have an issue with it,
  ```
if ('OPTIONS' === req.method) {
    res.send(200);
  }
```
will save you.

enriquezrene replied:
nice

oemel09 commented:
I updated the code and implemented the changes you proposed. Thanks for that!
Options is not yet implemented I think, so I'm not sure if I'll need that yet.

enriquezrene commented on manage-people/routes/PeopleRoutes.js:
```diff
@@ -2,20 +2,20 @@
             Person.scan().exec(function (err, data) {
                 if(err) {
-                    return res.status(500).send({message: 'Doh...'});
+                    return res.status(500).send({message: err});
```
I think providing technical details to end users is a big lack of security, that's why I was sending a Doh... instead. Can we keep doing that or do we have a reason to send the error.

oemel09 replied:
Okay, so I'll change it back to Doh again.

ralden replied:
Actually the end users of TinyUI are us; all super technical. A nice solution would be to have a systematic logging solution and log tons of error details on any error and display a URL that opens a nice log browser. That of course would be terrible for a real end user UI but it would be great for Tiny UI. As for "security"; to logging system should be protected but the URL that appears in the UI does not have to reveal anything more than a guid.
